tests/integration_tests.rs

```diff
@@ -622,6 +622,32 @@ fn complex_parallel_for_appender_loop() {
     unsafe { weld_value_free(ret_value) };
 }
 
+fn simple_for_vectorizable_loop() {
```

Comment:
Also we should have unit tests for the other builders that support vectorization.